zone/task_client_state.cpp

```diff
@@ -2212,42 +2212,23 @@ void ClientTaskState::RemoveTaskByTaskID(Client *client, uint32 task_id)
 	switch (task_type) {
```

Comment:
You can probably remove the switch/case entirely now and just check the task id without using the `task_type` variable

Remove the database calls above since that's handled by the `remove_from_db` passed to CancelTask now. Also remove the now unused `character_id` var.

Reply:
I agree with the DB calls and character_id.

I think we still need the switch statement, because the logic to get the sequence number passed to CancelTask changes based on task_type.